docs(website): help troubleshoot issues with multiple parsers (#5612)
typescript-eslint/typescript-eslint: docs/linting/TROUBLESHOOTING.md

};
```

## I am running into errors when parsing TypeScript in my .vue files

If you are running into issues parsing .vue files, it might be because parsers like [vue-eslint-parser](https://www.npmjs.com/package/vue-eslint-parser) are required to parse `.vue` files.  In this case you can move `@typescript-eslint/parser` inside `parserOptions` and use `vue-eslint-parser` as the top level parser.

```diff
- "parser": "@typescript-eslint/parser",
+ "parser": "vue-eslint-parser",
  "parserOptions": {
+     "parser": "@typescript-eslint/parser",
      "sourceType": "module"
  }
```

> The `parserOptions.parser` option can also specify an object to specify multiple parsers. See the [vue-eslint-parser README](https://github.com/vuejs/vue-eslint-parser#readme) for more details.

Reviewer: The docs link you chose to link in the PR description is also really good: https://eslint.vuejs.org/user-guide/#usage. If that's the right place to send folks, maybe we should send them there instead?

```suggestion
The `parserOptions.parser` option can also specify an object to specify multiple parsers. See the [vue-eslint-parser usage guide](https://eslint.vuejs.org/user-guide/#usage) for more details.
```

Across a large codebase, these can add up, and severely impact performance.

We recommend not using this rule, and instead using a tool like [`prettier`](https://www.npmjs.com/package/prettier) to enforce a standardized formatting.

> ## I am running into errors when parsing Typescript in my .vue files

Reviewer: Nit:

```suggestion
## I am running into errors when parsing TypeScript in my .vue files
```

Author: thanks!

> ## I am running into errors when parsing Typescript in my .vue files

Reviewer: A previous entry on the page mentions Vue already: _I use a framework (like Vue) that requires custom file extensions, ..._. What do you think about moving this entry to just after that other one, to group Vue entries together?

Reasoning: if someone is reading about Vue, they probably want to see both FAQs.

Author: good idea!

If you are running into issues parsing .vue files, it might be because parsers like [vue-eslint-parser](https://www.npmjs.com/package/vue-eslint-parser) are required to parse `.vue` files.  In this case you can move `@typescript-eslint/parser` inside `parserOptions` and use `vue-eslint-parser` as the top level parser.

```diff
- "parser": "@typescript-eslint/parser",
+ "parser": "vue-eslint-parser",
  "parserOptions": {
+     "parser": "@typescript-eslint/parser",
      "sourceType": "module"
  }
```

> The `parserOptions.parser` option can also specify an object to specify multiple parsers. See [vue-eslint-parser README](https://github.com/vuejs/vue-eslint-parser#readme) for more details.

Reviewer: Nit:

```suggestion
The `parserOptions.parser` option can also specify an object to specify multiple parsers. See the [vue-eslint-parser README](https://github.com/vuejs/vue-eslint-parser#readme) for more details.
```

Author: thanks
